mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -171,18 +171,18 @@ abstract class PredictionModel[FeaturesType, M <: PredictionModel[FeaturesType,
    * @param dataset input dataset
    * @return transformed dataset with [[predictionCol]] of type [[Double]]
    */
-  override def transform(dataset: DataFrame): DataFrame = {
+  override def transform(dataset: Dataset[_]): DataFrame = {
```

Comment:
What about the return type? If I want to chain transformers by `andThen`, i.e. `(tok.transform _).andThen(hashTF.transform)`, won't `DataFrame` be an issue? Why do we keep `DataFrame` as the return type?